### ISSUE TITLE
Opening a project sometimes results in a CPS fault in the StartupProjectRegistrar

### ISSUE THREAD

**Issue by srivatsn** [dotnet/roslyn-project-system#514]
Open a console app a few times. Sometimes, it results in a CPS fault with this stack:

```
9/27/2016 6:08:15 PM
Recoverable
System.FormatException: Unrecognized Guid format.
   at System.Guid.GuidResult.SetFailure(ParseFailureKind failure, String failureMessageID, Object failureMessageFormatArgument, String failureArgumentName, Exception innerException)
   at System.Guid.TryParseGuid(String g, GuidStyles flags, GuidResult& result)
   at System.Guid..ctor(String g)
   at Microsoft.VisualStudio.ProjectSystem.VS.Debug.StartupProjectRegistrar.<InitializeCoreAsync>d__14.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.VisualStudio.ProjectSystem.VS.Debug.StartupProjectRegistrar.<OnProjectFactoryCompletedAsync>d__13.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.VisualStudio.ProjectSystem.UnconfiguredProjectImpl.AutoLoadMethodStateMachine.<<StartExecution>b__8_0>d.MoveNext()
```

**Comment by 333fred** [comment]
This sounds like an issue with the ImplicitProjectPropertiesProvider again. Maybe a race between when the ProjectGuid property gets set and this code gets called?

**Comment by mavasani** [comment]
A quick fix to unblock rest of the team would be to first TryParseGuid before attempting to create a new Guid with the value read from project properties.

**Comment by davkean** [comment]
Yep, but not the longer workaround. This thing needs to set the correct GUID.

**Comment by basoundr** [comment]
@srivatsn I have not been able to repro this.

**Comment by basoundr** [comment]
is anyone else hitting this?

**Comment by mavasani** [comment]
I am hitting this way too often - almost 70-80% of all my debug sessions. Can you just try F5 latest Project-System to dev hive and close/open a net core project multiple times?

**Comment by abpiskunov** [comment]
I saw this issue too with web projects, so it seems to be generic issue for core and web... @basoundr  investigating it on my machine now

**Comment by basoundr** [comment]
Fixed by #555
